option.go

```diff
@@ -38,6 +38,14 @@ func OptionPrefix(x string) Option {
 	}
 }
 
+// OptionLivePrefix to set a dynamic prefix/prompt string. This option accepts a func() string which is then called on each redraw.
+func OptionLivePrefix(f func() string) Option {
```

Comment:
I think the options provided by this file should be used as arguments of `prompt.New` constructor function. So the state initialized by this functional options should be immutable. But `p.renderer.livePrefix` value is changed dynamically after running prompt.
If it allows to call this function after `prompt.Run` or `prompt.Input` , it will be more complex to control the terminal state.

And it seems that some of people confuse the difference between `OptionPrefix` and `OptionLivePrefix` . 
(ex) If both options are specified, which prefix option is enabled when starting go-prompt?

So we need to reconsider the interface to change prefix string dynamically.

Reply:
To identify which prefix we display, I think following interface is needed.

```go
// OptionLivePrefixFunc to set prefix string by callback.
func OptionLivePrefixFunc(f func(p *Prompt) (prefix string, useLivePrefix bool)) Option {
    return func(p *Prompt) error {
        p.livePrefixCallback = f
        return nil
    }
}
```